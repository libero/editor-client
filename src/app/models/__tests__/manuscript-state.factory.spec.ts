import {
  createAbstractState,
  createKeywordsState,
  createNewKeywordState,
  createTitleState
} from '../manuscript-state.factory';

describe('Manuscript state factory', () => {
  it('creates title state', () => {
    const el = document.createElement('article-title');
    el.innerHTML = 'A sample <italic>text</italic> with <sub>a</sub> variety of <sup>tags</sup>';
    const editorState = createTitleState(el);
    expect(editorState).toMatchSnapshot();
  });

  it('creates abstract state', () => {
    const el = document.createElement('abstract');
    el.innerHTML = 'A sample <italic>text</italic> with <sub>a</sub> variety of <sup>tags</sup>';
    const editorState = createAbstractState(el);
    expect(editorState).toMatchSnapshot();
  });

  it('creates keywords state', () => {
    const kwdContainer = document.createElement('div');
    kwdContainer.innerHTML = `<kwd-group kwd-group-type="author-keywords">
        <kwd>cerebellum</kwd>
        <kwd>climbing fiber</kwd>
        <kwd>reinforcement learning</kwd>
        <kwd><italic>vitae</italic></kwd>
        <kwd>Purkinje cells</kwd>
      </kwd-group>
      <kwd-group kwd-group-type="research-organism">
        <title>Research organism</title>
        <kwd>Mouse</kwd>
      </kwd-group>`;

    const editorState = createKeywordsState(Array.from(kwdContainer.querySelectorAll('kwd-group')));
    expect(editorState).toMatchSnapshot();
  });

  it('creates empty keyword state', () => {
    expect(createNewKeywordState()).toMatchSnapshot();
  });
});
