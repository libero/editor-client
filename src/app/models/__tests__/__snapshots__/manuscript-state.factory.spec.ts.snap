// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manuscript state factory creates abstract state 1`] = `
Object {
  "doc": Object {
    "content": Array [
      Object {
        "content": Array [
          Object {
            "content": Array [
              Object {
                "text": "A sample ",
                "type": "text",
              },
              Object {
                "marks": Array [
                  Object {
                    "type": "em",
                  },
                ],
                "text": "text",
                "type": "text",
              },
              Object {
                "text": " with ",
                "type": "text",
              },
              Object {
                "marks": Array [
                  Object {
                    "type": "sub",
                  },
                ],
                "text": "a",
                "type": "text",
              },
              Object {
                "text": " variety of ",
                "type": "text",
              },
              Object {
                "marks": Array [
                  Object {
                    "type": "sup",
                  },
                ],
                "text": "tags",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "abstract",
      },
    ],
    "type": "doc",
  },
  "selection": Object {
    "anchor": 2,
    "head": 2,
    "type": "text",
  },
}
`;

exports[`Manuscript state factory creates empty keyword state 1`] = `
Object {
  "doc": Object {
    "type": "keyword",
  },
  "selection": Object {
    "anchor": 0,
    "head": 0,
    "type": "text",
  },
}
`;

exports[`Manuscript state factory creates keywords state 1`] = `
Object {
  "author-keywords": Array [
    Object {
      "doc": Object {
        "content": Array [
          Object {
            "text": "cerebellum",
            "type": "text",
          },
        ],
        "type": "keyword",
      },
      "selection": Object {
        "anchor": 0,
        "head": 0,
        "type": "text",
      },
    },
    Object {
      "doc": Object {
        "content": Array [
          Object {
            "text": "climbing fiber",
            "type": "text",
          },
        ],
        "type": "keyword",
      },
      "selection": Object {
        "anchor": 0,
        "head": 0,
        "type": "text",
      },
    },
    Object {
      "doc": Object {
        "content": Array [
          Object {
            "text": "reinforcement learning",
            "type": "text",
          },
        ],
        "type": "keyword",
      },
      "selection": Object {
        "anchor": 0,
        "head": 0,
        "type": "text",
      },
    },
    Object {
      "doc": Object {
        "content": Array [
          Object {
            "marks": Array [
              Object {
                "type": "em",
              },
            ],
            "text": "vitae",
            "type": "text",
          },
        ],
        "type": "keyword",
      },
      "selection": Object {
        "anchor": 0,
        "head": 0,
        "type": "text",
      },
    },
    Object {
      "doc": Object {
        "content": Array [
          Object {
            "text": "Purkinje cells",
            "type": "text",
          },
        ],
        "type": "keyword",
      },
      "selection": Object {
        "anchor": 0,
        "head": 0,
        "type": "text",
      },
    },
  ],
  "research-organism": Array [
    Object {
      "doc": Object {
        "content": Array [
          Object {
            "text": "Mouse",
            "type": "text",
          },
        ],
        "type": "keyword",
      },
      "selection": Object {
        "anchor": 0,
        "head": 0,
        "type": "text",
      },
    },
  ],
}
`;

exports[`Manuscript state factory creates title state 1`] = `
Object {
  "doc": Object {
    "content": Array [
      Object {
        "content": Array [
          Object {
            "text": "A sample ",
            "type": "text",
          },
          Object {
            "marks": Array [
              Object {
                "type": "em",
              },
            ],
            "text": "text",
            "type": "text",
          },
          Object {
            "text": " with ",
            "type": "text",
          },
          Object {
            "marks": Array [
              Object {
                "type": "sub",
              },
            ],
            "text": "a",
            "type": "text",
          },
          Object {
            "text": " variety of ",
            "type": "text",
          },
          Object {
            "marks": Array [
              Object {
                "type": "sup",
              },
            ],
            "text": "tags",
            "type": "text",
          },
        ],
        "type": "article-title",
      },
    ],
    "type": "doc",
  },
  "selection": Object {
    "anchor": 1,
    "head": 1,
    "type": "text",
  },
}
`;
